keycloak/client.go

```diff
@@ -235,7 +244,12 @@ func (c Client) ListGroups(ctx context.Context) ([]Group, error) {
 		return nil, err
 	}
 
-	res := flatGroups(groups)
+	rootGroups := c.filterTreeWithRoot(groups)
+	if rootGroups == nil {
+		return nil, fmt.Errorf("could not find root group %q", c.RootGroup)
+	}
```

Comment:
Couldn't this also mean that there is the provided root group, but it does not have any subgoups (Or what exactly does it mean if `g.SubGroups == nil` :see_no_evil: )?  In that case we shouldn't return an error

Reply:
Omph now I got confused myself. There was indeed a bug there. It is now tested and fixed.